app/controllers/links_controller.rb

```diff
@@ -0,0 +1,43 @@
+class LinksController < ApplicationController
+  def edit
+    @local_authority = LocalAuthority.find_by(slug: params[:local_authority_slug])
+    @service = Service.find_by(slug: params[:service_slug])
+    @interaction = Interaction.find_by(slug: params[:interaction_slug])
+
+    @link = Link.get_link(params)
+    if @link.nil?
+      @link = Link.new
+      @link.url = 'n/a'
```

Comment:
+1 for keeping the 'n/a' out of the controller

Further to @davidbasalla's comment about `first_or_create` it might be interesting to make `Link.get_link` *always* return a `Link` instance.  In the case where the provided arguments allow us to fetch something from the DB the behaviour remains the same, but when those args don't resolved to a row in the DB we could build an unsaved `Link` instance and set it up with appropriate `LocalAuthority` and `ServiceInteraction` associations (e.g. what we do in the `create` action) so that we *could* save it once we've filled in the url.

I don't know that this is a great idea, but I think it's interesting, and it's a pattern I've used before.